src/shared_modules/rsync/src/registrationController.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Wazuh RSYNC
+ * Copyright (C) 2015, Wazuh Inc.
+ * May 26, 2022.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+*/
+
+
+#ifndef _REGISTRATIONCONTROLLER_HPP
+#define _REGISTRATIONCONTROLLER_HPP
+
+
+#include "commonDefs.h"
+#include <map>
+#include <shared_mutex>
+#include <string>
+
+
+class RegistrationController final
+{
+    private:
+        std::map<std::string, RSYNC_HANDLE> m_componentStatus;
+        std::shared_timed_mutex m_mutex;
+
+    public:
+        RegistrationController() = default;
+        // LCOV_EXCL_START
+        virtual ~RegistrationController() = default;
+        // LCOV_EXCL_STOP
+
+        void initComponentByHandle(const RSYNC_HANDLE handle,
+                                   const std::string& component)
+        {
+            std::lock_guard<std::shared_timed_mutex> lock(m_mutex);
+            m_componentStatus[component] = handle;
+        }
+
+        void removeComponentByHandle(const RSYNC_HANDLE handle)
+        {
+            std::lock_guard<std::shared_timed_mutex> lock(m_mutex);
+            auto it { m_componentStatus.begin() };
+
+            while (it != m_componentStatus.end())
+            {
+                if (it->second == handle)
+                {
+                    it = m_componentStatus.erase(it);
+                }
+                else
+                {
+                    ++it;
+                }
```

Comment:
There's a scenario whether multiples components share the rsync handler?
If the answer is yes, this LGTM. Otherwise, the next code could be improved to
```suggestion
                if (it->second == handle)
                {
                    m_componentStatus.erase(it);
                    break;
                }          
                ++it;
```

Reply:
> There's a scenario whether multiples components share the rsync handler?

Yes sir.